marlowe-playground-client/src/Simulation.purs

```diff
@@ -1,384 +1,260 @@
-          , onClick $ const $ Just
-              $ AddInput person (IDeposit accountId party token value) []
-          ]
-          [ text "+"
-          ]
-      ]
-    <> (renderDeposit accountId party token value)
-
-renderDeposit :: forall p. AccountId -> Party -> Token -> BigInteger -> Array (HTML p HAction)
-renderDeposit (AccountId accountNumber accountOwner) party tok money =
-  [ spanText "Deposit "
-  , b_ [ spanText (show money) ]
-  , spanText " units of "
-  , b_ [ spanText (show tok) ]
-  , spanText " into Account "
-  , b_ [ spanText (show accountOwner <> " (" <> show accountNumber <> ")") ]
-  , spanText " as "
-  , b_ [ spanText (show party) ]
-  ]
-
-inputChoice :: forall p. Boolean -> Maybe PubKey -> Int -> ChoiceId -> ChosenNum -> Array Bound -> HTML p HAction
-inputChoice isEnabled person index choiceId@(ChoiceId choiceName choiceOwner) chosenNum bounds =
-  let
-    validBounds = inBounds chosenNum bounds
+inputItem isEnabled person (DepositInput accountId party token value) =
+  li [ classes [ ClassName "deposit-a", aHorizontal ] ]
+    [ button
+        [ classes [ plusBtn, smallBtn ]
+        , enabled isEnabled
+        , onClick $ const $ Just
+            $ AddInput (Just person) (IDeposit accountId party token value) []
+        ]
+        [ text "+" ]
+    , p_ (renderDeposit accountId party token value)
+    ]
 
-    errorRow = if validBounds then [] else [ text boundsError ]
-  in
-    div_
-      ( [ button
-            [ class_ $ ClassName "composer-add-button"
-            , enabled isEnabled
-            , onClick $ const $ Just
-                $ AddInput person (IChoice (ChoiceId choiceName choiceOwner) chosenNum) bounds
-            ]
-            [ text "+"
-            ]
-        , spanText "Choice "
+inputItem isEnabled person (ChoiceInput choiceId@(ChoiceId choiceName choiceOwner) bounds chosenNum) =
+  li
+    [ classes [ ClassName "choice-a", aHorizontal ] ]
+    [ button
+        [ classes [ plusBtn, smallBtn ]
+        , enabled (isEnabled && inBounds chosenNum bounds)
+        , onClick $ const $ Just
+            $ AddInput (Just person) (IChoice (ChoiceId choiceName choiceOwner) chosenNum) bounds
+        ]
+        [ text "+" ]
+    , p [ class_ (ClassName "choice-input") ]
+        [ spanText "Choice "
         , b_ [ spanText (show choiceName) ]
         , spanText ": Choose value "
         , marloweActionInput isEnabled (SetChoice choiceId) chosenNum
         ]
-          <> errorRow
-      )
+    , p_ error
+    ]
   where
+  error = if inBounds chosenNum bounds then [] else [ text boundsError ]
+
   boundsError = "Choice must be between " <> intercalate " or " (map boundError bounds)
 
   boundError (Bound from to) = show from <> " and " <> show to
 
-inputNotify ::
-  forall p.
-  Boolean ->
-  Maybe PubKey ->
-  Int ->
-  HTML p HAction
-inputNotify isEnabled person index =
-  div_
+inputItem isEnabled person NotifyInput =
+  li
+    [ classes [ ClassName "choice-a", aHorizontal ] ]
```

Comment:
At least, the class name in here is misleading, because it is not a choice, not sure whether it is the right class

Reply:
it's not used actually.